framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -271,8 +278,14 @@ public void setRoot(/*@Nullable*/ CompilationUnitTree root) {
      * @return a tree annotator
      */
     protected TreeAnnotator createTreeAnnotator() {
-        return new ListTreeAnnotator(
-                new PropagationTreeAnnotator(this), new ImplicitsTreeAnnotator(this));
+        List<TreeAnnotator> treeAnnotators = new ArrayList<>();
+        treeAnnotators.add(new PropagationTreeAnnotator(this));
+        treeAnnotators.add(new ImplicitsTreeAnnotator(this));
+        if (expressionAnnotationHelper != null) {
+            treeAnnotators.add(
+                    new ExpressionAnnotationTreeAnnotator(this, expressionAnnotationHelper));
```

Comment:
Instead of hard-coding the class here, it would be nicer to ask the expressionAnnotationHelper for the right ExpressionAnnotationTreeAnnotator. That way, one could override the TreeAnnotator in a subclass of the helper.